src/aiodynamo/utils.py

```diff
@@ -197,3 +201,43 @@ def deparametetrize(
     for key, value in params.names.items():
         expression = expression.replace(key, value)
     return expression
+
+
+@dataclass(frozen=True)
+class MinLen2AppendOnlyList(Generic[T]):
+    first: T
+    second: T
```

Comment:
`first` and `second` are not used outside of this dataclass.

How about the following instead:
```py
@dataclass
class Foo:
    data: tuple

    @property
    def first(self): return self.data[0]

    @property
    def second(self): return self.data[1]
```

Reply:
they're there because this _must_ have at least two elements. it's illegal to create an instance with zero or one elements. that's why `first` and `second` are outside the tuple. they're implicitly used via `__iter__` [here](https://github.com/hennge/aiodynamo/blob/ce6e63f07da562c9a646d3a4c9a35512d8a80d6d/src/aiodynamo/expressions.py#L424) and [here](https://github.com/hennge/aiodynamo/blob/ce6e63f07da562c9a646d3a4c9a35512d8a80d6d/src/aiodynamo/expressions.py#L432)

Reply:
That can be solved by an assert or a ValueError in the `encode` method, can it not?

Reply:
it could at runtime, the goal here was to ensure it statically.